api/fastapi_service/services.py

```
@@ -674,161 +658,57 @@ def filter_by_polygon(polygon, edges, points):
+    with engine.connect() as conn:
+        result = conn.execute(query, {"city_id": city_id}).fetchall()
+        return [row[0] for row in result]
 
-def calc_metrics(points, edges):
+async def calc_metrics(points, edges, oneway_ids):
     points_list = [point[0] for point in points]
-    edges_list = [(edge[2], edge[3]) for edge in edges]
-
+    edges_list = [(edge[2], edge[3]) for edge in edges]  # Ориентированные ребра
+    reversed_edges_list = [(edge[3], edge[2]) for edge in edges if edge[1] not in oneway_ids]  # Обратные ребра для двусторонних дорог
     print(f"{datetime.now()} calc metrics begin")
```

Ruff (0.8.2)

`datetime` may be undefined, or defined from star imports

(F405)